spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/config/GcpConfigProperties.java

```diff
@@ -32,25 +35,46 @@
 @ConfigurationProperties("spring.cloud.gcp.config")
 public class GcpConfigProperties implements CredentialsSupplier {
 
-	/** Enables Spring Cloud GCP Config. */
+	/**
+	 *  Enables Spring Cloud GCP Config.
+	 */
 	private boolean enabled;
 
-	/** Name of the application. */
+	/**
+	 *  Name of the application.
+	 */
+	@Value("${spring.application.name:application}")
 	private String name;
 
-	/** Profile under which app is running (e.g., "prod", "dev", "test", etc.). */
-	private String profile = "default";
+	/**
+	 *  Comma-delimited string of profiles under which the app is running.
+	 */
+	private String profile;
 
-	/** Timeout for Google Runtime Configuration API calls. */
+	/**
+	 *  Timeout for Google Runtime Configuration API calls.
+	 */
 	private int timeoutMillis = 60000;
 
-	/** Overrides the GCP project ID specified in the Core module. */
+	/**
+	 *  Overrides the GCP project ID specified in the Core module.
+	 */
 	private String projectId;
 
-	/** Overrides the GCP OAuth2 credentials specified in the Core module. */
+	/**
+	 * Overrides the GCP OAuth2 credentials specified in the Core module.
+	 */
 	@NestedConfigurationProperty
 	private final Credentials credentials = new Credentials(GcpScope.RUNTIME_CONFIG_SCOPE.getUrl());
 
+	public GcpConfigProperties(Environment environment) {
+		String[] profiles = environment.getActiveProfiles();
+		if (profiles.length == 0) {
+			profiles = environment.getDefaultProfiles();
+		}
+		this.profile = StringUtils.arrayToCommaDelimitedString(profiles);
```

Comment:
So, you might end up with configuration name like `myapp_prod,xyz,perf` for the config name?
Are commas even allowed in the config name.
I think we're trying to be too smart here. I would just collapse everything into a simple `spring.cloud.gcp.config.name` like you and Ray suggested in the issue. Why are we not doing that here?

Reply:
Because [this is exactly what Spring Cloud Config Client is doing](https://github.com/spring-cloud/spring-cloud-config/blob/master/spring-cloud-config-client/src/main/java/org/springframework/cloud/config/client/ConfigClientProperties.java#L109), and I love that they solved this issue for us.